### PR TITLE
Enable automerge for appstudio-staging-ci PRs

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-1600

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

**Comment by lcarva** [comment]
To make this safer, we have to setup a branch protection rule that states the "pre-merge-ci" must pass. Otherwise, it gets merged before that completes.

File: .github/workflows/automerge.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - id: automerge
+        name: automerge
+        uses: "pascalgn/automerge-action@v0.15.5"
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          # Don't require a special label to auto merge.
+          MERGE_LABELS: ""
+          MERGE_FILTER_AUTHOR: appstudio-staging-ci
```
Comment (zregvart) [inline .github/workflows/automerge.yaml:32]: I think based on [this logic](https://github.com/pascalgn/automerge-action/blob/eb68b061739cb9d81564f8e812d0b3c45f0fb09a/lib/merge.js#L61) and a [JSON of the pull request](https://api.github.com/repos/hacbs-contract/ec-policies/pulls/279), this should be:

```suggestion
          MERGE_FILTER_AUTHOR: "appstudio-staging-ci[bot]"
```

**Review by zregvart — COMMENTED** [review]
Left two comments, one about the user filter is probably needs addressing. One nitplick would be to pass through `yq` to format like the other workflows.

File: .github/workflows/automerge.yaml (hunk 1 above)
Comment (lcarva) [reply .github/workflows/automerge.yaml]: Thanks! Updated.

File: .github/workflows/automerge.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - ready_for_review
+      - reopened
+      - unlocked
+  pull_request_review:
+    types:
+      - submitted
+  check_suite:
+    types:
+      - completed
+  status: {}
```
[the remark this answers is not in the document]
Comment (lcarva) [reply .github/workflows/automerge.yaml]: I took a different approach by leveraging the [built-in GitHub auto-merge](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-auto-merge-for-pull-requests-in-your-repository) functionality.

The new workflow just toggles the "enable auto-merge" button the PR. GitHub does the rest. I think it's a more elegant solution. I did have to enable auto-merges on the repo settings:

image

**Comment by lcarva** [comment]
FYI, I reverted the branch protection rule to require the branches to be up-to-date. I don't think we need that right now. Here's what it is now.
image

**Review by simonbaird — APPROVED** [review]
Nice.

**Comment by lcarva** [comment]
I'll go ahead and merge this. I'll keep an eye out for activity to make sure it's behaving correctly.

**Pull request merged**